10.0.4 release notes draft

pmaier1 opened the pull request:
Draft for 10.0.4 release notes.
- Still need input on some points (e.g. the Amazon v3 deprecation and some more. Please check the changes!)
- Need to add doc links at some points
- We might want to add known issues

@settermjd might want to help with formatting (looks a little strange right now :) ). 

@PVince81 @tomneedham

butonic commented on admin_manual/release_notes.rst:
```diff
@@ -12,6 +13,61 @@ Release Notes
+The option "Restrict users to only share with users in their groups" in the admin sharing settings forces users to only be able to share with groups they are member of while simultaneously prohibiting to share with single users that do not belong to any of the users' groups. To make this more granular we split this option into two and added "Restrict users to only share with groups they are membe ...
+
+- **Configurable solution for indistinguishable user display names**
+In the ownCloud sharing dialog users are displayed according to their display name. As users can choose their display name in self-service (can be disabled in config.php) and display names are not unique it is possible that a user can't distinguish sharing results. To cover this case the displayed user identifiers are now configurable. Administrators can configure to additionally display mail add ...
```
Link?

mmattel replied:
> Please check if this new feature better suits your needs.

Same comment as above

pmaier1 replied:
Link to what? Added a note that admins find the new setting in sharing settings. 

> Same comment as above

This is a completely new feature that might be of interest to check for administrators.

butonic commented on admin_manual/release_notes.rst:
```diff
@@ -12,6 +13,61 @@ Release Notes
+- **All amazon locations support v4 now - v3 deprecated**
+Need input on relevance of deprecation for existing external storages and on description of this item @tomneedham
+
+- **Modified time value of files is now 64 bits long**
```
mtime = modification time

mmattel replied:
All amazon locations --> **All Amazon locations

pmaier1 replied:
changed

butonic commented on admin_manual/release_notes.rst:
```diff
@@ -12,6 +13,61 @@ Release Notes
+**It is recommended to configure CRON to execute this background job regularly (it is not included in the regular ownCloud background jobs to give administrators more flexibility in scheduling)**. Please check the ownCloud documentation for more information. **LINK (https://github.com/owncloud/documentation/pull/3522)**
+
+- **Administrators can now exclude files from integrity check in config.php**
+When administrators did intentional changes to the ownCloud code they now have the ability to exclude certain files from the integrity checker.
```
AKA how to shoot yourself in the foot ...

PVince81 replied:
now that you mention it I'm not 100% that it works that way.

the original purpose was to exclude non-OC files (often server specific hidden files) that people have lying around inside the OC folder.

need to test whether excluding already signed files from integrity check works.

mmattel replied:
Agree to @PVince81 
would not state this as the reason for having that implemented which is different to this text.
(Do not open Pandoras Box...)
Please see the link to get an impression of: https://github.com/owncloud/core/pull/29126/files
Please reference to the documentation

PVince81 replied:
I have tested this with the 10.0.4RC1 tarball and can confirm that you can exclude patched files from integrity check as well as any new file that you might have created. So the description is good.

butonic commented on admin_manual/release_notes.rst:
```diff
@@ -12,6 +13,61 @@ Release Notes
+Need input on relevance of deprecation for existing external storages and on description of this item @tomneedham
+
+- **Modified time value of files is now 64 bits long**
+Note: When upgrading to 10.0.4 migrations may increase update duration dependent on number of files.
```
can't this be done without downtime ... we really should think about a strategy for doing this ... probably with a new table that gets renamed after the column has been changed ... this will take hours on large installations

pmaier1 replied:
not the right place to ask

butonic commented on admin_manual/release_notes.rst:
```diff
@@ -12,6 +13,61 @@ Release Notes
+Note: When upgrading to 10.0.4 migrations may increase update duration dependent on number of files.
+
+- **Updated minimum supported browser versions**
+Note: Users with outdated browsers might get warnings.
```
exact versions are?

phil-davis replied:
Stable10 PR https://github.com/owncloud/core/pull/29507
```
-		vs: { i: 10, f: 45, o: 0, s: 8, c: 45 },
+		vs: { i: 10, f: 54, o: 0, s: 9, c: 60 },
```
those are the max unsupported version.

phil-davis replied:
Doc update PR https://github.com/owncloud/documentation/pull/3505
```
+- Edge (current version on Windows 10)
 - IE11+ (except Compatibility Mode)
-- Firefox 14+
-- Chrome 18+
-- Safari 5+
+- Firefox 55+ or 52 ESR
+- Chrome 61+
+- Safari 10+
 
```

mmattel replied:
Please link to the documetation

pmaier1 replied:
Thanks, link added.

settermjd commented on admin_manual/release_notes.rst:
```diff
@@ -12,6 +13,61 @@ Release Notes
+Changes in 10.0.4
+-----------------
+
+Dear ownCloud administrator, please find below the changes and known issues of ownCloud Server 10.0.4 that need your attention:
```
How about "known issues in ownCloud Server" instead of "known issues of ownCloud Server"?

mmattel replied:
👍

pmaier1 replied:
changed

settermjd commented on admin_manual/release_notes.rst:
```diff
@@ -12,6 +13,61 @@ Release Notes
+**The full ownCloud Server 10.0.4 changelog can be found here: https://github.com/owncloud/core/blob/stable10/CHANGELOG.md**
+
+* **More granular sharing restrictions**
+The option "Restrict users to only share with users in their groups" in the admin sharing settings forces users to only be able to share with groups they are member of while simultaneously prohibiting to share with single users that do not belong to any of the users' groups. To make this more granular we split this option into two and added "Restrict users to only share with groups they are membe ...
```
s/they are member/they are a member/g

phil-davis replied:
Note: the text on the UI says "...they are member of"
one day, maybe, that could be changed to say "...they are a member of"

mmattel replied:
> Please check if the new possibilities better suit your needs.

Is imho not necessary, as the old way still exists and the new implementation is an addition

pmaier1 replied:
thanks, changed. as @phil-davis said the option is called like this in the UI..

settermjd commented on admin_manual/release_notes.rst:
```diff
@@ -12,6 +13,61 @@ Release Notes
+The option "Restrict users to only share with users in their groups" in the admin sharing settings forces users to only be able to share with groups they are member of while simultaneously prohibiting to share with single users that do not belong to any of the users' groups. To make this more granular we split this option into two and added "Restrict users to only share with groups they are membe ...
+
+- **Configurable solution for indistinguishable user display names**
+In the ownCloud sharing dialog users are displayed according to their display name. As users can choose their display name in self-service (can be disabled in config.php) and display names are not unique it is possible that a user can't distinguish sharing results. To cover this case the displayed user identifiers are now configurable. Administrators can configure to additionally display mail add ...
```
How about the following:

---
The ownCloud sharing dialog displays users according to their display name. As users can choose their display name in self-service (which can be disabled in `config.php`) and display names are not unique, it is possible that a user can't distinguish sharing results. To cover this case the displayed user identifiers are now configurable. Administrators can now configure the display of either mail addresses or user ids. Please check if this new feature better suits your needs.
---

tomneedham commented:
Fix bug when viewing folder with > 1000 files in https://github.com/owncloud/core/pull/29416

tomneedham commented:
Amazon v3 signature is already depreceated in some of their regions. We updated our usage to always use v4 signature, which is now supported at all regions

PVince81 commented:
Note: the release notes are not the same as changelog. General bugfixes belong in the changelog.

Release notes are entries worthy of getting admin's attention:
- things that might motivate an admin to upgrade
- things that might prevent an admin to upgrade (known issues with some systems that might be in use)
- things that need attention and action after an update (ex: must setup a new cron)

settermjd commented on admin_manual/release_notes.rst:
```diff
@@ -12,6 +13,61 @@ Release Notes
+
+- **Added "occ files:scan" repair mode to repair filecache inconsistencies**
+We recommend to use this command when directed to do so in the upgrade process (TODO more info)
+Please check the ownCloud documentation for more information. **LINK (https://github.com/owncloud/documentation/issues/3446)**
```
I'd suggest directly integrating the link:

Please check [the ownCloud documentation](https://github.com/owncloud/documentation/issues/3446) for more information.

mmattel replied:
👍

pmaier1 replied:
This is a placeholder for the correct doc link. I actually don't want to link to a ticket/PR.

settermjd commented on admin_manual/release_notes.rst:
```diff
@@ -12,6 +13,61 @@ Release Notes
+Please check the ownCloud documentation for more information. **LINK (https://github.com/owncloud/documentation/issues/3446)**
+
+- **Detailed mode for "occ security:routes"**
+Administrators can use the output of this command when using a network firewall to check the appropriateness of configured rules or to get assistance when setting up.
```
I suggest adding a little more punctuation, e.g., 

"Administrators can use the output of this command when using a network firewall, to check the appropriateness of configured rules or to get assistance when setting up."

mmattel replied:
when setting up --> when setting it up

pmaier1 replied:
changed

settermjd commented on admin_manual/release_notes.rst:
```diff
@@ -12,6 +13,61 @@ Release Notes
+Administrators can use the output of this command when using a network firewall to check the appropriateness of configured rules or to get assistance when setting up.
+
+- **Added mode of operations to differentiate between single-instance or clustered setup**
+As ownCloud needs to behave differently when operating in a clustered setup we added a config.php option ``'operation.mode' => 'single-instance' or 'clustered-instance'``. 
```
For utmost clarity, I'd write this a little differently:

As ownCloud needs to behave differently when operating in a clustered setup to a single instance setup, we added a new `config.php` option `operation.mode`. It can take one of two values: `single-instance` and `clustered-instance`.
For example:

```
'operation.mode' => 'single-instance',
```

mmattel replied:
clustered setup to a single instance setup --> clustered setup versus a single instance setup

we added a new ``config.php`` option ``operation.mode`` --> a new ``config.php`` option ``operation.mode`` has been added

pmaier1 replied:
changed

settermjd commented on admin_manual/release_notes.rst:
```diff
@@ -12,6 +13,61 @@ Release Notes
+As ownCloud needs to behave differently when operating in a clustered setup we added a config.php option ``'operation.mode' => 'single-instance' or 'clustered-instance'``. 
+Currently the Market App (ownCloud Marketplace integration) does not support clustered setups and can do harm when used for installing or updating apps. The new config setting prevents this and other actions that are undesired in cluster mode.
+**Therefore when operating in a clustered setup it is mandatory to set this option.**
+Please check the ownCloud documentation for more information. **LINK (https://github.com/owncloud/documentation/pull/3506)**
```
As above, enable the link directly in the text:

---

Please check [the ownCloud documentation](https://github.com/owncloud/documentation/pull/3506) for more information.

mmattel replied:
👍

pmaier1 replied:
This is a placeholder for the correct doc link. I actually don't want to link to a ticket/PR.

settermjd commented on admin_manual/release_notes.rst:
```diff
@@ -12,6 +13,61 @@ Release Notes
+Please check the ownCloud documentation for more information. **LINK (https://github.com/owncloud/documentation/pull/3506)**
+
+- **Added occ dav:cleanup-chunks command to clean up expired uploads**
+When file uploads are interrupted by any reason already uploaded file parts (chunks) stay on the storage to be able to resume the file upload in a following upload run. Resuming an upload is only possible until the partial upload is expired and deleted, respectively. To cleanup chunks (expire and delete) originating from unfinished uploads administrators can use this newly introduced command. The ...
```
I'd suggest some minor changes, as below:

---

When file uploads are interrupted for any reason, already uploaded file parts (chunks) remain in the underlying storage so that the file upload can resume in a future upload attempt. 

However, resuming an upload is only possible until the partial upload is either expired or deleted. To clean up chunks (expire and delete) originating from unfinished uploads, administrators can use this newly introduced command. The expiry time default is two days, but it can be specified as a parameter to the command.

mmattel replied:
The expiry time default is two days --> The default expiry time is two days, but can be specified as an additional  parameter to this command.

Pls reference to the documentation

pmaier1 replied:
Changed. No doc available.

settermjd commented on admin_manual/release_notes.rst:
```diff
@@ -12,6 +13,61 @@ Release Notes
+
+- **Added occ dav:cleanup-chunks command to clean up expired uploads**
+When file uploads are interrupted by any reason already uploaded file parts (chunks) stay on the storage to be able to resume the file upload in a following upload run. Resuming an upload is only possible until the partial upload is expired and deleted, respectively. To cleanup chunks (expire and delete) originating from unfinished uploads administrators can use this newly introduced command. The ...
+**It is recommended to configure CRON to execute this background job regularly (it is not included in the regular ownCloud background jobs to give administrators more flexibility in scheduling)**. Please check the ownCloud documentation for more information. **LINK (https://github.com/owncloud/documentation/pull/3522)**
```
As above, enable the link directly in the text:

---

**It is recommended to configure Cron to execute this background job regularly.** It is not included in the regular ownCloud background jobs so that the administrators have more flexibility in scheduling it. Please check [the ownCloud documentation](https://github.com/owncloud/documentation/pull/3522) for more information.

mmattel replied:
Cron --> CRON (preferred) or ``cron``

pmaier1 replied:
changed

pmaier1 commented:
Thanks for feedback. Incorporated suggested changes.

> Amazon v3 signature is already depreceated in some of their regions. We updated our usage to always use v4 signature, which is now supported at all regions

This does not help me. Please tell me if this is relevant in the release notes and what exactly needs to be addressed by an admin.

pmaier1 commented:
Don't merge yet:
- Doc link missing for "Added "occ files:scan" repair mode to repair filecache inconsistencies" (not yet merged)
- The known issues list is empty which is a good thing of course. Please add if there is something.

mmattel commented:
Merged: https://github.com/owncloud/core/pull/28253 (Add repair step to repair mismatch filecache paths)
Open: https://github.com/owncloud/core/pull/29581 (Update the help output for the occ files:scan command)

settermjd commented:
@pmaier1 is this ready to merge?

pmaier1 commented:
- Added doc link for occ files:scan repair mode
- Added hint that there are no known issues (I want to prevent people wondering if we missed to mention it in the release notes)

> Open: owncloud/core#29581 (Update the help output for the occ files:scan command)

@mmattel You mention in the PR that this would be important for the release notes. I don't see the relation. Can you explain?

Apart from this I think we're good here. Anyway, I would wait for the release to merge and publish it.

mmattel commented:
I only added in the referenced open core PR that it would be good to have that one also being ready before 10.0.4 gets released to have things proper and clean documented. The text text provided here is fine to me 👍

PVince81 commented on admin_manual/release_notes.rst:
```diff
@@ -12,6 +13,65 @@ Release Notes
+
+**Known issues**
+
+Currently there are no known issues for ownCloud Server 10.0.4.
```
Should we remove this section ? This seems to state that ownCloud 10.0.4 is bug-free which is not correct 😄

settermjd replied:
As no software's bug-free, I'd say yes.

pmaier1 replied:
As I said above I left the section not to confuse people that we might have missed to add known issues. This way it states that there are no issues that we consider as "known" and "to-be-fixed". Should we remove it?

settermjd replied:
No, I'd leave it. It's clear in what it says, and it's honest.

PVince81 commented on admin_manual/release_notes.rst:
```diff
@@ -12,6 +13,65 @@ Release Notes
+Note: Users with outdated browsers might get warnings. See the list of supported browser versions: https://doc.owncloud.com/server/10.0/admin_manual/installation/system_requirements.html#web-browser
+
+- **10.0.3 known issues are solved**
+  - SFTP external storages with key pair mode work again
```
should we add links to resolved tickets ?

settermjd replied:
Yep. It's great for transparency.

pmaier1 replied:
Added.

mmattel commented:
@PVince81 should a note be added that https://github.com/owncloud/core/pull/29261 (Remove double quotes from boolean values in status.php output) has been solved ?

PVince81 commented:
> @PVince81 should a note be added that owncloud/core#29261 (Remove double quotes from boolean values in status.php output) has been solved ?

Considering that it was a known issue in 10.0.3 it might make sense to mention that it's fixed now, yes.

pmaier1 commented:
> @PVince81 should a note be added that owncloud/core#29261 (Remove double quotes from boolean values in status.php output) has been solved ?

Added.

PVince81 commented:
Ok I only see thumbs up everywhere, so it seems this is ready for merge then ?

PVince81 commented:
@settermjd does this need a backport to become visible ? If yes, can you take care ? Thanks

settermjd commented:
On it now.

settermjd commented:
Done!
